Sources/SwiftFrameCore/Workers/AttributedStringCache.swift

```diff
@@ -0,0 +1,88 @@
+import AppKit
+import Foundation
+
+// Used for keeping attributed strings that were parsed on the main thread in memory for asynchronous processing
+final class AttributedStringCache: NSObject {
+
+    // MARK: - Nested Type
+
+    enum FontMode {
+        case dynamic(maxSize: CGFloat)
+        case fixed(pointSize: CGFloat)
+    }
+
+    // MARK: - Properties
+
+    static let shared = AttributedStringCache()
+    private lazy var cache = NSCache<NSString, NSAttributedString>()
```

Comment:
**question** - should this be `NSCache` ? It is subject to eviction when memory resources are low.

https://developer.apple.com/documentation/foundation/nscache